engine/src/witness/btc/source.rs

```diff
@@ -45,8 +45,12 @@ where
 					loop {
 						tick.tick().await;
 
+						let best_block_hash = client
+							.best_block_hash()
+							.await
+							.expect("TODO: This whole source will be deleted ");
```

Comment:
This is just waiting on the ingress-egress tracker it seems. @MxmUrw you were looking into that right?

Reply:
Yes it's on my list, but haven't specifically started yet.